src/SOS/SOS.Extensions/DebuggerServices.cs

```diff
@@ -478,6 +480,40 @@ public HResult GetLastException(out uint processId, out int threadId, out EXCEPT
             return hr;
         }
 
+        public void FlushCheck()
+        {
+            VTable.FlushCheck(Self);
+        }
+
+        public ImmutableArray<string> ExecuteHostCommand(string commandLine, DEBUG_OUTPUT interestMask = DEBUG_OUTPUT.NORMAL)
```

Comment:
Any reason we are opting for ImmutableArray here over say IReadOnlyList? Is there anything that depends on CoW semmantics?

Reply:
What CoW (I assume you mean copy-on-write) semantics does ImmutableArray have? I just want an array that the elements can't be changed/written.  Why would IReadOnlyList be better?

Reply:
I think I see now.  List<T> already implements IReadOnlyList<T> and no copying/conversion would have to happen.

Reply:
I'll change it to IReadOnlyList<string>.

Reply:
ImmutableArray allows you to write - and it just allocates a new array. It just doesn't modify the old one. Things like https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutablearray-1.add?view=net-8.0. So if I don't expect that to be a concern - I just prefer IReadOnly...

Reply:
ImmutableArrays have lower overhead than IReadOnlyList, or other similar constructs.

IMO though, it actually wasn't worthwhile to use ImmutableArray in ClrMD.  The dependency alone is a headache to maintain (as you see from our conversation about 6.0 vs 8.0), the perf benefit isn't noticeable (especially in something as slow as debugging), and the contract guarantees of immutability don't make things easier.  In retrospect, just returning `ClrModule[]` instead of `ImmutableArray<ClrModule>` would have been fine.  But that ship has sailed for ClrMD to make that big of a change without a lot of churn.

Avoiding immutable is a good decision.  Even just returning raw arrays is conditionally fine, if it doesn't hurt anything if a user decides to modify one.  Just my 2 cents.